build.gradle

```diff
@@ -68,6 +68,8 @@ apiValidation {
             "integration",
     ]
 
+    nonPublicMarkers += ["kotlinx.benchmark.internal.KotlinxBenchmarkRuntimeInternalApi"]
```

Comment:
Most likely, I'm missing a bit of understanding here, but if, instead of making some API internal or private, we decided to keep it public for a while and only warn its potential users, then it means we do care about binary compatibility and this API should not be filtered from dumps.

And if we don't care about binary compatibility, then we are probably free to make that API internal/private.

Reply:
Yes, normally the process is to gradually increment from an opt-in warning, then to an opt-in error, and then remove it. That's not the case here.

@qurbonzoda and I discussed the process regarding the new Gradle plugin opt-in annotation, and we couldn't find any public usage of should-be-internal functions, so we're agreed to skip some steps.

I'm not familiar with the benchmark-runtime API, so maybe the same approach isn't suitable here.

Reply:
I would expect we either make these functions private/internal and remove them (well, they will be removed automatically) from dumps, or we keep them public and keep them in dumps as well.

Reply:
I see it the following way: 
Currently, we do not provide source or binary compatibility guarantees for the library. We try to make the migration process smooth because we care about our users. 
All APIs annotated with @InternalApi will soon be internal. Thus, going forward, we want to track changes to other public APIs. Again, this doesn’t mean we aren’t going to break compatibility; it's rather to have visibility on what is changing in the public API. Here, those unintentionally public APIs would distort the picture.

Reply:
So, here are two independent efforts going on:
First, the gradual removal of unintentionally public APIs.
Second, keeping track of intentionally public APIs.

Reply:
Does that make sense to you?
If not, I prefer to keep them in dumps for the time being, rather than making them internal right away.

Reply:
> Currently, we do not provide source or binary compatibility guarantees for the library.

With that in mind, I don't mind the change. Thanks for elaborating!